lib/zabbix/zabbix.go

```diff
@@ -0,0 +1,145 @@
+package zabbix
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// API struct details for Zabbix
+type API struct {
+	URL      string
+	Key      string
+	User     string
+	Password string
+}
+
+// Payload struct holds info needed for an API call
+type Payload struct {
+	Method  string      `json:"method"`
+	Params  interface{} `json:"params,omitempty"`
+	Auth    string      `json:"auth"`
+	JSONRPC string      `json:"jsonrpc"`
+	ID      int         `json:"id"`
+}
+
+// Result struct holds a Zabbix host info
+type Result struct {
+	HostID      string `json:"hostid"`
+	Host        string `json:"host"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+// Results struct holds a list of Zabbix host info
+type Results []Result
+
+// NewAPI builds new API struct
+func NewAPI(URL, key string) API {
```

Comment:
Just a thought ... would it be worth adding a validation here to ensure these values are not empty (if they can't be) 🤔

Reply:
good idea.